### PR TITLE
Dynamic fall rate

### PR DESCRIPTION
**Pull request by AdityaHarvi**
Dynamic fall rate based off current level & spectator voting.
Max level is set to be 15.

Upon loading into SceneGameArena, the client will request the current fall rate from the server. Once the value is received, the timer will be modified on the client and the blocks will fall at the new value.

I was wondering if this may cause sync issues across the different clients but with the "source of truth" being worked on, Idk how much of an issue that actually is.

### CODE REVIEW

**Comment by swebra** [comment]
Looks good to me, and looks like this covers level increase logic as well. Are the level increase communications already implemented or is that future work to be done?

**Comment by AdityaHarvi** [comment]
> Looks good to me, and looks like this covers level increase logic as well. Are the level increase communications already implemented or is that future work to be done?

Level is already implemented and hooked up to the scoreboard logic

File: server/src/Level.ts [hunk 1]
```diff
@@ -1,36 +1,110 @@
...
+    /**
+     * Check if its possible to update the game's level based off the given score. Max level is 15.
+     * @param score The total score the players have accumulated.
+     */
     public checkUpdateLevel(score: number) {
-        if (score >= this._currentLevel * 20) {
+        if (
+            score - (this._currentLevel - 1) * 20 >= this._currentLevel * 20 &&
+            this._currentLevel < 15
+        ) {
```
Comment (swebra) [inline server/src/Level.ts:38]: This still doesn't comply with the SRS due to score rollover (as alluded to in my previous comment)

From SRS 4.8.3:
> Additional points scored above a given level’s goal do not contribute to the goal of the next level.

**Review by swebra — CHANGES_REQUESTED** [review]

File: server/src/Level.ts (hunk 1 above)
Comment (AdityaHarvi) [reply server/src/Level.ts]: I've renamed the _totalScore in the Scoreboard.ts file to now be _accumualtedScore.
It represents the accumulated score for the current level and is reset whenever the level increases.

**Review by swebra — APPROVED** [review]
LGTM

**Pull request merged**